pkg/cable/vxlan/vxlan.go

```diff
@@ -0,0 +1,538 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package vxlan
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/admiral/pkg/log"
+	v1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/cable"
+	"github.com/submariner-io/submariner/pkg/natdiscovery"
+	"github.com/submariner-io/submariner/pkg/types"
+	"github.com/submariner-io/submariner/pkg/util"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	"k8s.io/klog"
+)
+
+const (
+	VxlanIface             = "vxlan-tunnel"
+	VxlanOverhead          = 50
+	VxlanVTepNetworkPrefix = 241
+	CableDriverName        = "vxlan"
+	TableID                = 100
+)
+
+type Operation int
+
+const (
+	Add Operation = iota
+	Delete
+	Flush
+)
+
+type vxlan struct {
+	localEndpoint types.SubmarinerEndpoint
+	connections   []v1.Connection
+	mutex         sync.Mutex
+	vxlanIface    *vxlanIface
+	nattPort      int `default:"4500"`
+}
+
+type vxlanIface struct {
+	activeEndpointHostname string
+	link                   *netlink.Vxlan
+	vtepIP                 net.IP
+}
+
+type vxlanAttributes struct {
+	name     string
+	vxlanID  int
+	group    net.IP
+	srcAddr  net.IP
+	vtepPort int
+	mtu      int
+}
+
+func init() {
+	cable.AddDriver(CableDriverName, NewDriver)
+}
+
+func NewDriver(localEndpoint types.SubmarinerEndpoint, localCluster types.SubmarinerCluster) (cable.Driver, error) {
+	var err error
+
+	v := vxlan{
+		localEndpoint: localEndpoint,
+	}
+
+	port, err := localEndpoint.Spec.GetBackendPort(v1.UDPPortConfig, int32(v.nattPort))
```

Comment:
`v.nattPort` will be 0 here - it seems you want a default value of 4500 since you've defined a field tag as such. But there must be some code to parse the tag and set the field but I don't see that here.

Reply:
Made the logic simpler using a constant.